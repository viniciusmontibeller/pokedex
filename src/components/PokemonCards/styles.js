import styled from 'styled-components'

const Card = styled.li`
    background-color: ${props => { props.theme.defaultValues.transparentBackground}};
    position: relative;
    box-shadow: ${props => props.theme.defaultValues.darkGlassShadow};
    backdrop-filter: ${props => props.theme.defaultValues.backgroundFilter};
    width: 280px;
    color: ${props => props.theme.theme.color};
    text-align: center;
    padding: 15px;
    border-radius: 10px;
    transition: ${props => props.theme.defaultValues.transition};
    letter-spacing: 1px;

    img{
        filter: drop-shadow(0px 0px 10px white);
        transition: ${props => props.theme.defaultValues.transition};
    }

    ::after{
        content: '';
        position: absolute;
        height: 80%;
        width: 100%;
        border-radius: inherit;
        z-index: -1;
        inset: 0;
        background-color: ${props => props.theme.pokemonTypeColor[props.colorType]};
        box-shadow: ${props => props.theme.defaultValues.shadow};
        opacity: 0.6;
        transition: ${props => props.theme.defaultValues.transition};
    }

    :hover{
        opacity: 1;
        backdrop-filter: none;
        scale: 1.08;
        ::after{
            opacity: 0.9;
        }
        img{
        filter: drop-shadow(0px 0px 15px white);
        }
    }
`

const Name = styled.p`
    text-transform: capitalize;
    margin-bottom: 15px;
    font-weight: 400;
    font-size: 25px;
`

const Id = styled.p`
    text-align: left;
    border-radius: 50%;
`

const Img = styled.img`
    height: 150px;
    margin-bottom: 10px;
`

export { Card, Name, Id, Img }