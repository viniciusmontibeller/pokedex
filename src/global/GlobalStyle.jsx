import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    *{
        margin: 0;
        padding: 0;
        box-sizing: border-box;
    }

    ul {
        list-style: none;
    }

    a {
        text-decoration: none;
    }

    body{
        background: url(${props => props.theme.theme.backgroundImage}) no-repeat center center fixed; 
        background-size: cover;
        background-color: ${props => props.theme.theme.backgroundColor};
        font-family: ${props => props.theme.defaultValues.fontFamily};

        #root {
            height: auto;
        }
    }
`

export { GlobalStyle }