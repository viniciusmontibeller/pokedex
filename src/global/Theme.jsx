import React, { useEffect } from 'react'
import { ThemeProvider } from 'styled-components'
import { useState } from 'react'
import LightBackground from '../assets/lightbg.jpg'
import DarkBackground from '../assets/darkbg.jpg'

const themes = {
    light: {
        color: "rgba(34, 34, 34, 1)",
        backgroundColor: "rgba(255, 255, 255, 1)",
        backgroundColorHover: "rgba(34, 34, 34, 0.2)",
        backgroundImage: LightBackground,
    },
    dark: {
        color: "rgba(255, 255, 255, 1)",
        backgroundColor: "rgba(34, 34, 34, 1)",
        backgroundColorHover: "rgba(255, 255, 255, 0.3)",
        backgroundImage: DarkBackground,
    }
};

const defaultValues = {
    transparentBackground: "rgba(255, 255, 255, 0.1)",
    glassShadow: "hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset, 0 0 10px 2px rgba(0, 0, 0, 0.35)",
    darkGlassShadow: "hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset, 0 0 10px 5px rgba(0, 0, 0, 0.35)",
    shadow: "0 0 10px 1px rgba(0, 0, 0, 0.35)",
    hoverShadow: "0 0 10px 5px rgba(0, 0, 0, 0.35)",
    transition: "0.2s ease-in-out",
    backgroundFilter: "blur(8px)",
    fontFamily: "'Nunito', sans-serif"
}

const pokemonTypeColor = {
    fire: "#EE8130",
    grass: "#7AC74C",
    electric: "#F7D02C",
    water: "#6390F0",
    ground: "#E2BF65",
    rock: "#B6A136",
    fairy: "#D685AD",
    poison: "#A33EA1",
    bug: "#A6B91A",
    dragon: "#6F35FC",
    psychic: "#F95587",
    flying: "#A98FF3",
    fighting: "#C22E28",
    normal: "#A8A77A",
    ice: "#96D9D6",
    ghost: "#735797",
    dark: "#705746",
    steel: "#B7B7CE",
}

const localTheme = JSON.parse(localStorage.getItem('theme')) || themes.light

const Theme = ({children}) => {
        const [ theme, setTheme ] = useState(localTheme)

        useEffect(() => {
            localStorage.setItem('theme', JSON.stringify(theme))
        }, [theme])

        return (
        <ThemeProvider theme={{theme, setTheme, pokemonTypeColor, defaultValues}}>
            {children}
        </ThemeProvider>
        )
}

export { Theme, themes }