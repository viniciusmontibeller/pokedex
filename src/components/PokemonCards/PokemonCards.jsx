import { Types, Type } from '../Types'
import { Card, Img, Id, Name } from './styles'

const PokemonCard = ({ name, image, types, id }) => {

    return (
        <Card colorType={types[0].type.name}>
            <Id># {id}</Id>
            <Img src={image} alt={name} />
            <Name>{name}</Name>
            <Types>
                {types.map((type, index) => {
                    return <Type
                        type={type.type.name}
                        key={index}>
                        {type.type.name}
                    </Type>
                }
                )}
            </Types>
        </Card>
    )
}

export { PokemonCard }