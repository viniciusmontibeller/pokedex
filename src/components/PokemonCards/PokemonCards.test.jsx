import { PokemonCard } from "./PokemonCards";
import { render } from "@testing-library/react";
import { Theme } from "../../global/Theme";

describe('Pokemon Card component', () => {
    const pokemonFake = {
        id: '6',
        name: 'charizard',
        image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png',
        types:
            [
                {
                    "slot": 1,
                    "type": {
                        "name": "grass",
                        "url": "https://pokeapi.co/api/v2/type/12/"
                    }
                },
                {
                    "slot": 2,
                    "type": {
                        "name": "poison",
                        "url": "https://pokeapi.co/api/v2/type/4/"
                    }
                }
            ]
        
    }
    it('should render a pokemon card with info', () => {
        const {asFragment} = render(<Theme><PokemonCard {...pokemonFake}/></Theme>)
        expect(asFragment).toMatchSnapshot()
    })
})