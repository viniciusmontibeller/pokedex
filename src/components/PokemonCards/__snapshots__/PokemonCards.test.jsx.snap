// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card component should render a pokemon card with info 1`] = `[Function]`;
